cmd/up.go

```diff
@@ -15,17 +15,81 @@
 package cmd
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/Sirupsen/logrus"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
 var upCmd = &cobra.Command{
 	Use:   "up",
 	Short: "Start KDK container",
-	Long: `Start KDK container`,
+	Long:  `Start KDK container`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("up called")
+		logger := logrus.New().WithField("command", "up")
+
+		imageCoordinates := []string{viper.Get("image.repository").(string), viper.Get("image.tag").(string)}
+
+		client, err := client.NewEnvClient()
+		if err != nil {
+			logger.WithField("error", err).Fatal("Failed to create docker client")
+		}
+
+		ctx := context.Background()
+
+		var binds []string
+
+		for _, bind := range viper.Get("docker.binds").([]interface{}) {
+			configBind := make(map[string]string)
```

Comment:
the hard cast to string already assumes the key is a string so this map is maybe not needed.
I believe the body of this for could be one line.
https://play.golang.org/p/TsdEnhaiiX1